src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -135,6 +140,10 @@ def _on_config_reload(self) -> None:
             self._thread = threading.Thread(target=self.__discover_cluster, daemon=True)
             self._thread.start()
 
+        if not hasattr(self, '_custom_resource_thread') and self.namespace_monitored_resources:
```

Comment:
I think running it in a single thread, introduces some race conditions and synchronisation issues (both in memory and when writing to DB)
I think we should just do it as part of discovery to avoid it.
I think it's fine, if discovery will be a little slower, once an hour - I don't expect this to add much latency